### PR TITLE
Multisig validation for slashing

### PR DESCRIPTION
**Pull request by scolear**

### CODE REVIEW

**Review by Rayerleier — COMMENTED** [review]
Add a slashIndex as a salt value in the signature encoding.
Add slashedInfos to record slash

File: test/iBTC_NetworkMiddleware.t.sol [hunk 1]
```diff
@@ -319,126 +332,108 @@ contract iBTC_NetworkMiddlewareTest is Test {
...
+        bytes32 messageHash = keccak256(dataToSign);
+
+        signatures = new bytes[](approvedSignerKeys.length);
+
+        for (uint256 i = 0; i < approvedSignerKeys.length; i++) {
+            signatures[i] = _signMessage(approvedSignerKeys[i], messageHash);
+        }
+
+        return signatures;
+    }
```
Comment (scolear) [inline test/iBTC_NetworkMiddleware.t.sol:431]: great job!

File: test/iBTC_NetworkMiddleware.t.sol [hunk 2]
```diff
@@ -319,126 +332,108 @@ contract iBTC_NetworkMiddlewareTest is Test {
...
-        );
-
-        assertEq(iBTC_delegator.stake(NETWORK.subnetwork(0), alice), 0);
 
         vm.prank(OWNER);
         iBTC_delegator.grantRole(NETWORK_LIMIT_SET_ROLE, alice);
         _setNetworkLimit(alice, NETWORK, networkLimit);
 
-        assertEq(iBTC_delegator.stake(NETWORK.subnetwork(0), alice), 0);
-
```
Comment (scolear) [inline test/iBTC_NetworkMiddleware.t.sol:365]: Why did you remove these check here?

File: src/iBTC_NetworkMiddleware.sol [hunk 3]
```diff
@@ -360,12 +377,14 @@ contract NetworkMiddleware is SimpleKeyRegistry32, Ownable {
                 uint256 vaultStake =
                     IBaseDelegator(IVault(vault).delegator()).stakeAt(subnetwork, operator, epochStartTs, new bytes(0));
                 _slashVault(epochStartTs, vault, subnetwork, operator, (amount * vaultStake) / totalOperatorStake);
+                slashedInfos[slashIndex++] =
+                    SlashedInfo({epoch: epoch, operator: operator, slashedAmount: amount, timeStamp: block.timestamp});
```
Comment (scolear) [inline src/iBTC_NetworkMiddleware.sol:381]: What exactly is the reason for this?

**Review by scolear — COMMENTED** [review]
LGTM!

**Review by Rayerleier — APPROVED** [review]
